--- a/src/hb-ot-font.cc
+++ b/src/hb-ot-font.cc
@@ -665,7 +665,7 @@ hb_ot_get_glyph_v_origins (hb_font_t *font,
 	hb_position_t origin;
 	unsigned cv;
 	if (origin_cache->get (*first_glyph, &cv))
-	  origin = font->y_scale < 0 ? -cv : cv;
+	  origin = font->y_scale < 0 ? -static_cast<hb_position_t>(cv) : static_cast<hb_position_t>(cv);
 	else
 	{
 	  origin = font->em_scalef_y (VORG.get_y_origin (*first_glyph));
@@ -689,7 +689,7 @@ hb_ot_get_glyph_v_origins (hb_font_t *font,
 	hb_position_t origin;
 	unsigned cv;
 	if (origin_cache->get (*first_glyph, &cv))
-	  origin = font->y_scale < 0 ? -cv : cv;
+	  origin = font->y_scale < 0 ? -static_cast<hb_position_t>(cv) : static_cast<hb_position_t>(cv);
 	else
 	{
 	  origin = font->em_scalef_y (VORG.get_y_origin (*first_glyph) +
@@ -727,7 +727,7 @@ hb_ot_get_glyph_v_origins (hb_font_t *font,
       hb_position_t origin;
       unsigned cv;
       if (origin_cache->get (*first_glyph, &cv))
-	origin = font->y_scale < 0 ? -cv : cv;
+	origin = font->y_scale < 0 ? -static_cast<hb_position_t>(cv) : static_cast<hb_position_t>(cv);
       else
       {
 	origin = font->em_scalef_y (glyf.get_v_origin_with_var_unscaled (*first_glyph, font, *scratch, gvar_cache));
@@ -761,7 +761,7 @@ hb_ot_get_glyph_v_origins (hb_font_t *font,
       unsigned cv;
 
       if (origin_cache->get (*first_glyph, &cv))
-	origin = font->y_scale < 0 ? -cv : cv;
+	origin = font->y_scale < 0 ? -static_cast<hb_position_t>(cv) : static_cast<hb_position_t>(cv);
       else
       {
 	hb_glyph_extents_t extents = {0};
